src/handlers/spec-rewriter.ts

```diff
@@ -47,80 +49,97 @@ export class SpecificationRewriter {
       },
     } = this.context;
 
-    const sysPromptTokenCount = countTokens(createSpecRewriteSysMsg([], UBIQUITY_OS_APP_NAME, ""));
-    const queryTokenCount = countTokens(llmQuery);
+    const sysPromptTokenCount = encode(createSpecRewriteSysMsg([], UBIQUITY_OS_APP_NAME, "")).length;
+    const queryTokenCount = encode(llmQuery).length;
 
-    const modelMaxTokenLimit = await this.context.adapters.openRouter.completions.getModelMaxTokenLimit();
-    const maxCompletionTokens = await this.context.adapters.openRouter.completions.getModelMaxOutputLimit();
+    const tokenLimit = await this.context.adapters.openRouter.completions.getModelTokenLimits();
 
-    if (!modelMaxTokenLimit || !maxCompletionTokens) {
+    if (!tokenLimit) {
       throw this.context.logger.error(`The token limits for configured model ${this.context.config.openRouterAiModel} were not found`);
     }
     const tokenLimits: TokenLimits = {
-      modelMaxTokenLimit,
-      maxCompletionTokens,
+      modelMaxTokenLimit: tokenLimit.contextLength,
+      maxCompletionTokens: tokenLimit.maxCompletionTokens,
       tokensRemaining: 0,
     };
-
     // what we start out with to include files
     tokenLimits.tokensRemaining = tokenLimits.modelMaxTokenLimit - tokenLimits.maxCompletionTokens - sysPromptTokenCount - queryTokenCount;
-    const githubConversation = await fetchIssueConversation(this.context, tokenLimits);
+    // reduce 10% to accomodate token estimate
+    tokenLimits.tokensRemaining = 0.9 * tokenLimits.tokensRemaining;
+    const githubConversation = await this.fetchIssueConversation(this.context, tokenLimits);
 
-    return await completions.createCompletion(openRouterAiModel, githubConversation, UBIQUITY_OS_APP_NAME, maxCompletionTokens);
+    return await completions.createCompletion(openRouterAiModel, githubConversation, UBIQUITY_OS_APP_NAME, tokenLimit.maxCompletionTokens);
   }
 
-  async getUserRole(context: Context) {
-    const orgLogin = context.payload.organization?.login;
-    const user = context.payload.sender.login;
-    const { logger, octokit } = context;
-
+  async canUserRewrite() {
+    if (this.context.payload.sender.type === "BOT") return true;
```

Comment:
```suggestion
    if (this.context.payload.sender.type === "Bot") return true;
```
`Bot` is the correct type